lib/config/presets/internal/replacements.ts

```diff
@@ -648,6 +649,17 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  'mem-rename': {
+    description: '`mem` was renamed to `memoize`.',
+    packageRules: [
+      {
+        matchDatasources: ['npm'],
+        matchPackageNames: ['mem'],
+        replacementName: 'memoize',
+        replacementVersion: '10.0.0',
```

Comment:
If someone is on v89 or lower, won't this be an unnecessarily breaking change?

Reply:
> I'm not sure I understood the question, hopefully this answers it:

Migrating to `memoize` v10 is always a breaking change, even from v9.
`mem` has versions up until 9 (no 10), and `memoize` has v10 (no v9 or lower).

Reply:
> > (I'm not sure I understood the question, hopefully this answers it)
> 
> Migrating to `memoize` v10 is always a breaking change, even from v9.

Right, so that's why this is a bad idea to replace unless unvoidable.

> `mem` has versions up until 9 (no 10), and `memoize` has v10 (no v9 or lower).

Not according to https://www.npmjs.com/package/mem, it says there's a 10.0.0

Reply:
Oh right! It is set as deprecated though, so Renovate will not offer v10 of `mem`.

Reply:
> Right, so that's why this is a bad idea to replace unless unvoidable.

I understand, is there any way to make it offer the `mem` v10 major, and then the `memoize` replacement only for that v10?

Reply:
Deprecated logic fix is in https://github.com/renovatebot/renovate/pull/27875